glean-core/ffi/src/lib.rs

```diff
@@ -37,17 +37,28 @@ lazy_static! {
 type RawStringArray = *const *const c_char;
 
 /// Create a vector of strings from a raw C-like string array
-unsafe fn from_raw_string_array(arr: RawStringArray, len: i32) -> Vec<String> {
-    if arr.is_null() || len == 0 {
-        return vec![];
+///
+/// ## Safety
+///
+/// * We check the array pointer for validity (non-null)
+/// * FfiStr checks each individual char pointer for validity (non-null)
+/// * We discard invalid char pointers (null pointer)
+/// * invalid UTF-8 is replaced with the replacement character (guaranteed by ffi-support)
+fn from_raw_string_array(arr: RawStringArray, len: i32) -> Vec<String> {
+    unsafe {
```

Comment:
So. philosophically, the bulk of this change is really just hiding the `unsafe` within the function, rather than doing anything that's inherently safer?  (Fine either way, just seems like an interesting "Rustism")...

Reply:
We can't skip unsafety completely, we're dealing with raw pointers here.
[As explained here](https://github.com/mozilla/glean/pull/179#discussion_r303951239) `unsafe` doesn't mean it's all actually unsafe, it's just necessary to explicitly deal with all the cases the compiler can't prove.